01-messaging.md

```diff
@@ -896,6 +901,27 @@ failure:
 1. Invalid stream: 0xffffffffffffffffff 00 00 00
 2. Reason: valid TLV records but invalid ordering
 
+## Appendix C: Message Extension
+
+This section contains examples of valid and invalid extensions on the `init`
+message. The base `init` message (without extensions) for these examples is
+`0x001000000000` (all features turned off).
+
+The following `init` messages are valid:
```

Comment:
What do you mean? Since that variable size byte slice starts with a `u16` that explicitly sets its length, I believe it can be extended. For the specific case of `init`, it is even already extended with a TLV stream since we added the `networks` TLV. Can you clarify if I misunderstood?

Reply:
Ah do you mean that the existing message may fill the `65535` bytes of the message? In that case definitely, it will hit the message size limit and fail. Do you think we should mention that the extension can only be done within the bounds of the message length limit?

Reply:
Yeah I mean the feature vectors can fill the entire message. Adding a mention/note of this sounds good to me. In the _typical_ case this shouldn't happen, but implementations should be ready for it.

Reply:
Great, I'll add that.

Reply:
Added in https://github.com/lightningnetwork/lightning-rfc/pull/754/commits/a58cece70f895603ee2deeadbc9f45ce7305d094.

Did you really implement that many hidden features in lnd that you're afraid of filling up the `init`? :laughing: